### PR TITLE
Daft: issue-18 Add temperature and topP input

### PR DESCRIPTION
**Pull request by Noppagorn**
Untitled

### CODE REVIEW

**Comment by dtinth** [comment]
Thank you! Unfortunately this breaks the mobile UI.

image

Maybe you can use a [split button](https://getbootstrap.com/docs/5.3/components/dropdowns/#split-button) that displays a form…

image

…or make the button bar wrap by adding `flex-wrap` class to the bar

image

**Comment by Noppagorn** [comment]
What about this? use split button.
Untitled3

**Review by dtinth — COMMENTED** [review]
The split button looks very good! Thank you so much.

Almost there — can you move the model selection into the split button? This way the text generation settings can be set from one place (in the drop down) rather than two.

File: src/App.tsx [hunk 1]
```diff
@@ -740,6 +789,42 @@ function useSelectedModel() {
   ]
 }
 
+function useTemperature(nodeId: string) {
+  const key = 'temperature/' + nodeId;
```
[the remark this answers is not in the document]
Comment (Noppagorn) [reply src/App.tsx]: OK updated (UI and global scope)
Untitled4

**Review by dtinth — APPROVED** [review]
@Noppagorn it’s looking great now, thank you very much krub.

**Pull request merged**